### PR TITLE
Build for rhel8

### PR DESCRIPTION
**Pull request by RMTT**
Build plcontainer for rhel8

### CODE REVIEW

File: concourse/pipeline.yml [hunk 1]
```diff
@@ -132,7 +133,7 @@ resources:
 - name: plcontainer_src
   type: git
   source:
-    branch: 6X_STABLE
+    branch: sa/6X_rhel8
```
Comment (beeender) [inline concourse/pipeline.yml:136]: branch name needs to be changed before merge

File: concourse/pipeline.yml [hunk 2]
```diff
@@ -1,18 +1,21 @@
 ################################################################################
-#################################### GROUPS ####################################
+#Bearer eyJhbGciOiJSUzI1NiIsImtpZCI6IiIsInR5cCI6IkpXVCJ9.eyJjc3JmIjoiM2Q2MTUwMTcxNWNiODk1MmE4MTljMmQzNmYyOGQ3M2ZjNDEyMWIxNzQwMTJhMzZhNWVhZmI5Mzg0NDNiZTY2MSIsImVtYWlsIjoiZC5yb25nQG91dGxvb2suY29tIiwiZXhwIjoxNjQyNjQ2NDU5LCJpc19hZG1pbiI6dHJ1ZSwibmFtZSI6IiIsInN1YiI6IkNnZ3lNRFU1TmpVNU9SSUdaMmwwYUhWaSIsInRlYW1zIjp7Im1haW4iOlsib3duZXIiXX0sInVzZXJfaWQiOiIyMDU5NjU5OSIsInVzZXJfbmFtZSI6IlJNVFQifQ.o6fwmMiXLT2Vxt3Y9q2BBvOyeWNOLyg88HY2W40UCO2JV9y347wdRpgL6TG_mXTSIs5ayKMzYXOug4UIgDRifL5jpLCzhxroQ1BVIxNVOy8bnY9w6XpwRnKChfPxd9SbCoKkanF7JO4MxFiPQNOC2fzXmRr2p0cHA5CDrXC3lFsb6wkyuPwgGlmlRNjkIwXKc21CcSLqupXlZWRmDZL_ER5qFUcnXCPCUKbh5Kvgo-eoaP8xI_nwbxxiE3wYcrJqsPaiguvMU4WlEci65dn8LCl-dWDSGDFkdLu_A8JbuQkpCDDBjZHrXjV1QCV3eJj0pFBBmB8gJQbkd9uFsNsmTA################################### GROUPS ####################################
```
Comment (beeender) [inline concourse/pipeline.yml:2]: why do we need this?

File: concourse/release-pipeline.yml [hunk 3]
```diff
@@ -110,14 +112,14 @@ resources:
 - name: plcontainer_src
   type: git
   source:
-    branch: 6X_STABLE
+    branch: sa/6X_STABLE
```
Comment (beeender) [inline concourse/release-pipeline.yml:115]: ditto

File: concourse/release-pipeline.yml [hunk 4]
```diff
@@ -110,14 +112,14 @@ resources:
...
     uri: https://github.com/greenplum-db/plcontainer.git
     tag_filter: 2.*
 
 - name: gpdb_src
   type: git
   source:
-    branch: 6X_STABLE
+    branch: sa/6X_STABLE
```
Comment (beeender) [inline concourse/release-pipeline.yml:122]: ditto

File: concourse/scripts/build_plcontainer_prepare.sh [hunk 5]
```diff
@@ -13,27 +13,37 @@ TOP_DIR=${CWDIR}/../../../
...
-  ${TOP_DIR}/gpdb_src/concourse/scripts/setup_gpadmin_user.bash "centos"
-  make_cluster
- 
+
+  if [ "${BLD_OS}" = "rhel8" ]; then
+    # build json-c staticly
+    git clone https://github.com/json-c/json-c.git
+    cd json-c
+    git fetch --all --tags
+    git checkout tags/json-c-0.15-20200726
```
Comment (beeender) [inline concourse/scripts/build_plcontainer_prepare.sh:24]: Replace 21-24 with
```
git clone -b tags/json-c-0.15-20200726 --depth 1 https://github.com/json-c/json-c.git 
```

To make it faster and save some bandwidth

File: concourse/scripts/run_plcontainer_resgroup_tests.sh [hunk 6]
```diff
@@ -11,6 +11,7 @@ else
 fi
 
 if [ "$TEST_OS" = centos7 -o "$TEST_OS" = sles12 -o "$TEST_OS" = ubuntu18 ]; then
+    #NOTE: not used
```
Comment (beeender) [inline concourse/scripts/run_plcontainer_resgroup_tests.sh:14]: not used for rhel8 only?

File: concourse/scripts/build_plcontainer_prepare.sh [hunk 7]
```diff
@@ -13,27 +13,37 @@ TOP_DIR=${CWDIR}/../../../
...
-  yum -y install epel-release
-  yum -y install R  
-
   # setup gpdb environment
   install_gpdb
-  ${TOP_DIR}/gpdb_src/concourse/scripts/setup_gpadmin_user.bash "centos"
-  make_cluster
- 
+
+  if [ "${BLD_OS}" = "rhel8" ]; then
```
Comment (beeender) [inline concourse/scripts/build_plcontainer_prepare.sh:19]: we should do static link for all releases.
Add a FIXME above the `if` and let's fix it later.

File: src/hidejsonc.ld.version [hunk 8]
```diff
@@ -0,0 +1,5 @@
+{
+local:
+  json_object_*;
+  json_tokener_*;
```
Comment (beeender) [inline src/hidejsonc.ld.version:4]: shall we use a whitelist instead? like:
```
{
global:
    pg_*;
    Pg_magic_func;
    ....
    _PG_init;
local:
    *;
};
```

File: concourse/pipeline.yml [hunk 9]
```diff
@@ -305,6 +336,30 @@ jobs:
...
+    - get: gpdb_src
+    - get: plcontainer_src
+    - get: plcontainer_client_ubuntu18_build
+      passed: [plcontainer_client_build]
+      trigger: true
+    - get: rhel-image-build-8
+  - task: plcontainer_gpdb_build
+    file: plcontainer_src/concourse/tasks/plcontainer_build.yml
+    params:
+      BLD_OS: rhel8
```
Comment (Sasasu) [inline concourse/pipeline.yml:353]: What's the meaning of `BLD_OS`, I think it is a typo from `PLC_OS`

File: concourse/scripts/build_plcontainer_prepare.sh (hunk 7 above)
Comment (Sasasu) [reply concourse/scripts/build_plcontainer_prepare.sh:19]: got compile error on retro gcc

File: concourse/pipeline.yml (hunk 9 above)
Comment (beeender) [reply concourse/pipeline.yml:353]: not sure if if `BLD_OS` means `BUILD_OS` where the package was build from.

File: concourse/scripts/build_plcontainer_prepare.sh (hunk 7 above)
Comment (beeender) [reply concourse/scripts/build_plcontainer_prepare.sh:19]: yea, just do the static link for rhel8 only for now. let's fix it later.

File: concourse/pipeline.yml (hunk 9 above)
Comment (RMTT) [reply concourse/pipeline.yml:353]: `BUILD_OS`

File: concourse/pipeline.yml (hunk 2 above)
Comment (RMTT) [reply concourse/pipeline.yml:2]: i don't know, maybe it's just a wrong paste.

File: concourse/scripts/build_plcontainer_prepare.sh (hunk 5 above)
Comment (RMTT) [reply concourse/scripts/build_plcontainer_prepare.sh:24]: `cd` is not needed?

File: concourse/release-pipeline.yml [hunk 10]
```diff
@@ -110,7 +112,7 @@ resources:
 - name: plcontainer_src
   type: git
   source:
-    branch: 6X_STABLE
+    branch: sa/6X_rhel8
```
Comment (beeender) [inline concourse/release-pipeline.yml:115]: change this before merge

File: concourse/scripts/build_plcontainer_prepare.sh (hunk 5 above)
Comment (beeender) [reply concourse/scripts/build_plcontainer_prepare.sh]: cd is still needed

**Review by beeender — APPROVED** [review]

**Pull request merged**